wordpress-mobile/WordPress-Android
# Issue/12028 refactor bookmark action

Parent issue #12028

This PR is targeting 15.7 but it can be merged even to 15.8. 

This PR refactors the BookmarkAction:
1. Replaces LiveData living inside a usecase with Flow
2. Moves snackbar logic outside of the usecase
3.  Removes logic related to refreshing the content on bookmark action from the ReaderPostListAdapter
4. Fixes tests
5. Adds tests related to bookmark action for the ReaderPostActionHandler

To test:
Test this action on following and discover tabs (enable RI FF if the PR which removes it hasn't been merged yet)

Dialog
1. Clear app data
2. Open Reader - Following tab
3. Click on Bookmark action on one of the items
4. Notice a dialog is shown
5. Click on OK and go to Saved tab
6. Notice the post is saved there

Snackbar
1. Open Reader - Following tab
1. Click on Bookmark action on another item
2. Notice Snackbar is shown
3. Click on "View All" action in the snackbar
4. Notice the "Saved" tab is shown and the post is there

Offline
1. Open Reader - Following tab
1. Click on Bookmark action on yet another item
2. Notice snackbar is shown
3. Wait for cca 10s
4. Turn on airplane mode
5. Go to Saved tab
6. Open the post and notice the content (even images) are loaded correctly even though you are offline

Unbookmark
1. Click on unbookmark action on one of the saved items
2. Notice the post gets unbookmarked and is removed from the saved tab (it might still be there but in a collapsed mode with "Undo" action)




PR submission checklist:
